### PR TITLE
Azure SQL Server auto discovery

### LINKED ISSUE #17062: Azure SQLServer auto discovery.
### What
Implement auto-discovery support for  Azure SQLServer instances.

### How: 
We already supporting azure auto discovery for `mysql`, `postgres` and `redis` azure instance.  The integration should use the same mechanism for SQLServer azure instance auto discovering. 

The azureMatcher should be extended https://github.com/gravitational/teleport/blob/d319ee2734287a44e1941d55b3c4d025581b8d96/lib/srv/db/cloud/watchers/watcher.go#L198 by SQLServer Fetcher allowing to fetch and configure current SQLServer instances. 

#### Accept Criteria: 
SQLServer instances are auto discovery by teleport.

### PR DESCRIPTION
**Pull request by gabrielcorado**
Closes #17062 by adding auto-discovery for Azure SQL servers.

 The authentication method used by the discovered servers is the [Azure Activity Directory](https://github.com/gravitational/teleport/pull/17142). This means no extra authentication steps are required for them.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -357,6 +357,8 @@ require (
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 )
 
+require github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/sql/armsql v1.0.0
```
Comment (greedy52) [inline go.mod:360]: I think we should put this in the existing block.

**Review by greedy52 — COMMENTED** [review]
First pass. Overall looks very clean =D.

Are we adding auto discovery for SQL Server to `teleport db configure create`?

File: lib/srv/db/cloud/watchers/azure_sql.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	client, err := cfg.AzureClients.GetAzureSQLServerClient(subID)
+	return client, trace.Wrap(err)
+}
+
+func (f *azureSQLServerFetcher) GetServerLocation(server *armsql.Server) string {
+	return azure.StringVal(server.Location)
+}
+
+func (f *azureSQLServerFetcher) NewDatabaseFromServer(server *armsql.Server, log logrus.FieldLogger) types.Database {
+	database, err := services.NewDatabaseFromAzureSQLServer(server)
```
[the remark this answers is not in the document]
Comment (smallinsky) [reply lib/srv/db/cloud/watchers/azure_sql.go:46]: I think that we should. Azure MySQL and Postgres does this in https://github.com/gravitational/teleport/blob/d319ee2734287a44e1941d55b3c4d025581b8d96/lib/srv/db/cloud/watchers/azure_redis.go#L7

File: lib/cloud/azure/sql_test.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+			tc.expectErr(t, err)
+			requireSQLServers(t, tc.expectedServers, servers)
+		})
+	}
+}
+
+func requireSQLServers(t *testing.T, expected []string, actual []*armsql.Server) {
+	t.Helper()
+
+	serverNames := []string{}
```
Comment (smallinsky) [inline lib/cloud/azure/sql_test.go:124]: nit: 
```suggestion
	var serverNames []string{}
```
https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices

File: lib/srv/db/cloud/watchers/azure_sql.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (gabrielcorado) [reply lib/srv/db/cloud/watchers/azure_sql.go:46]: I've added this check for Managed instances. For SQL Servers, only one field is related to the database status (`state`). While testing, I couldn't find a case where the API we're using to list SQL Servers returned a server with `status != "ready"`, so I'm unsure if it is worth adding this check.

It is also worth mentioning that this field doesn't have documentation. Also, deleted/deleting instances are not returned using this client, and updating instances still, return `status = "ready"`.

**Review by greedy52 — APPROVED** [review]

File: lib/services/database.go [hunk 4]
```diff
@@ -792,6 +855,26 @@ func labelsFromAzureRedisEnterprise(cluster *armredisenterprise.Cluster, databas
...
+	return withLabelsFromAzureResourceID(labels, azure.StringVal(server.ID))
+}
+
+// labelsFromAzureManagedSQLServer creates database labels from the provided
+// Azure Managed SQL server.
+func labelsFromAzureManagedSQLServer(server *armsql.ManagedInstance) (map[string]string, error) {
+	labels := azureTagsToLabels(azure.ConvertTags(server.Tags))
+	labels[types.OriginLabel] = types.OriginCloud
+	labels[labelRegion] = azure.StringVal(server.Location)
+	labels[labelEngineVersion] = AzureSQLManagedInstanceLabel
```
[the remark this answers is not in the document]
Comment (gabrielcorado) [reply lib/services/database.go]: It doesn't have the engine version field. I've removed this label.

**Comment by gabrielcorado** [comment]
@atburke @smallinsky Can you take a look, please?

**Review by smallinsky — COMMENTED** [review]
CI seems to complain about undefined `AzureSQLManagedInstanceLabel` variable so LGTM when it will be fixed.

**Review by smallinsky — APPROVED** [review]

**Pull request merged**